plugin/src/main/java/org/openrewrite/gradle/isolated/DefaultProjectParser.java

```diff
@@ -826,7 +826,7 @@ public Stream<SourceFile> parse(Project subproject, Set<Path> alreadyParsed, Exe
                         sourceSetSourceFiles = Stream.concat(
                                 sourceSetSourceFiles,
                                 omniParser.parse(accepted, baseDir, new InMemoryExecutionContext())
-                        );
+                        ).map(it -> it.withMarkers(it.getMarkers().add(javaVersion)));
```

Comment:
This should instead be on the `omniParser` as presently, it would be updating the entire set of source files. While in the simple case it may be a noop, for multiple source sets with differing JavaVersion objects, it would clobber the ones that came before it.

Reply:
Fixed via https://github.com/openrewrite/rewrite-gradle-plugin/commit/9992c581fea019e36b44ff7eb62bd10c11ff97bb.